### PR TITLE
Clean up legend docstrings

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

File: lib/matplotlib/legend.py [hunk 1]
```diff
@@ -1107,16 +1117,16 @@ def set_title(self, title, prop=None):
         self.stale = True
 
     def get_title(self):
-        'return Text instance for the legend title'
+        'Return `~.text,.Text` instance for the legend title.'
```
Comment (anntzer) [inline lib/matplotlib/legend.py]: pyto

File: lib/matplotlib/legend.py [hunk 2]
```diff
@@ -1140,12 +1150,16 @@ def get_bbox_to_anchor(self):
 
     def set_bbox_to_anchor(self, bbox, transform=None):
         """
-        set the bbox that the legend will be anchored.
+        Set the bbox that the legend will be anchored.
+
+        *bbox* can be
+
+        - A `BboxBase` instance
+        - A tuple of ``[left, bottom, width, height]`` in the given transform
```
Comment (anntzer) [inline lib/matplotlib/legend.py]: use parentheses for tuples? :)

File: lib/matplotlib/legend.py (hunk 1 above)
Comment (dstansby) [reply lib/matplotlib/legend.py]: Huh?

Comment (anntzer) [reply lib/matplotlib/legend.py]: typo :-)
(sorry that joke is getting old)

File: lib/matplotlib/legend.py [hunk 3]
```diff
@@ -1140,12 +1150,16 @@ def get_bbox_to_anchor(self):
 
     def set_bbox_to_anchor(self, bbox, transform=None):
         """
-        set the bbox that the legend will be anchored.
+        Set the bbox that the legend will be anchored.
+
+        *bbox* can be
+
+        - A `BboxBase` instance
```
Comment (jklymak) [inline lib/matplotlib/legend.py]: `~.BboxBase`

File: lib/matplotlib/legend.py [hunk 4]
```diff
@@ -1258,7 +1272,7 @@ def draggable(self, state=None, use_blit=False, update="loc"):
           * False : turn draggable off
 
         If draggable is on, you can drag the legend on the canvas with
-        the mouse.  The DraggableLegend helper instance is returned if
+        the mouse. The `DraggableLegend` helper instance is returned if
```
Comment (jklymak) [inline lib/matplotlib/legend.py]: `~.DraggableLegend`

**Comment by jklymak** [comment]
@dstansby I just pushed to your branch (which I'm 99.9% sure you said was OK in the past).  Feel free to revert if you don't like it.  I'll commit if the checks pass...

**Comment by dstansby** [comment]
I'm always happy for small doc fixes to be pushed to my branches, thanks a lot!

**Pull request merged**